### PR TITLE
add optional symbology to dataset

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Dataset-Info hat Default-Symbology](https://trello.com/c/HjsYyo9I/27-dataset-info-hat-default-symbology)

File: services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs [hunk 1]
```diff
@@ -167,6 +167,7 @@ impl SentinelS2L2aCogsDataProvider {
                             no_data_value: band.no_data_value,
                         }
                         .into(),
+                        symbology: None, // TODO
```
Comment (ChristianBeilschmidt) [inline services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs:170]: Can't you specify this 1-6000 or what this is, here?

File: services/src/datasets/storage.rs [hunk 2]
```diff
@@ -77,6 +81,7 @@ impl From<ImportDataset> for Dataset {
             description: value.description,
             result_descriptor: value.result_descriptor,
             source_operator: value.source_operator,
+            symbology: None,
```
Comment (ChristianBeilschmidt) [inline services/src/datasets/storage.rs:84]: We should have at least one test for the dataset and one for the handler I guess.

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs:170]: the colorizer is different for each band, so I can't easily specify it here

File: services/src/datasets/storage.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/datasets/storage.rs:84]: what do you want me to test?

Comment (ChristianBeilschmidt) [reply services/src/datasets/storage.rs:84]: We should test the case where we have specified a symbology and return it to the client.
Then, we know when something breaks in our return type.

File: services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs:170]: We can use 0-10k and implement an if-else-block later.
Currently, you don't see anything which isn't a working solution.

https://gis.stackexchange.com/questions/233874/what-is-the-range-of-values-of-sentinel-2-level-2a-images

File: services/src/datasets/storage.rs (hunk 2 above)
Comment (michaelmattig) [reply services/src/datasets/storage.rs:84]: do you mean checking the serialization of a Dataset? I can do it, but we already have tests for Symbology serialization

File: services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/pro/datasets/external/sentinel_s2_l2a_cogs.rs]: caca41556981504247dd1879887e075b91eedc47

**Pull request merged**